src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,203 @@
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCar(resultSet);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get car by id " + id, throwable);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversFromDb(car));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT cars.id AS id, model, manufacturer_id, name, country FROM cars "
+                + "JOIN manufacturers ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars.is_deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCar(resultSet));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get cars from DB.", throwable);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversFromDb(car));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? WHERE id = ? "
+                + "AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't update car " + car, throwable);
+        }
+        removeDriversFromCar(car);
+        assignDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() != 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete car with ID " + id, throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT car_id FROM cars_drivers WHERE driver_id = ?";
```

Comment:
we need  to know more than just id

Reply:
u can use where is_deleted = FALSE and don't use optional

Reply:
I just copy pasted the method signature from the task description.
Driver id is sufficient for the method to work. 

I'm re-using method 'get(Long id) here. It returns empty optional when there's no match for a given input.
get checks for is_deleted=FALSE.